### PR TITLE
Stable LoRA addition and webui text2video extension support.

### PR DESCRIPTION
**Pull request by ExponentialML**
This adds support for training LoRA files that will be compatible for use with the [webui extension](https://github.com/kabachuha/sd-webui-text2video).

The webui compatible LoRA files ***are not*** a variant of cloneofsimo's or kohya's, but instead uses the official implementation by Microsoft (with added Conv3d support). This choice was purely based on ease of use and compatibility between different repositories / projects. You can find the Stable Diffusion compatible implementation [here.](https://github.com/ExponentialML/Stable-LoRA)

This will be done before the text2video webui support, but will be in progress asynchronously with this PR until I open a PR at the extension repository.

- [x] Create a LoRA Handler that allows for training multiple implementations `[cloneofsimo, stable_lora]`.
- [x] Ensure the cloneofsimo variant works with the new LoRA Handler (only tested Stable LoRA).
- [x] Test that loading and unloading LoRAs during training are not regressed in anyway.
- [x] Update instructions of `.yaml` files as needed.

### CODE REVIEW

File: stable_lora/lora.py [hunk 1]
```diff
@@ -184,10 +184,11 @@ def forward(self, x: torch.Tensor):
         if self.r > 0 and not self.merged:
             return F.conv3d(
                 x, 
-                self.weight + (self.lora_B.transpose(0,1) @ self.lora_A.transpose(0,1)).view(self.weight.shape) * self.scaling,
-                self.bias, self.stride, self.padding, self.dilation, self.groups
+                self.weight + torch.mean((self.lora_B @ self.lora_A).view(self.view_shape), dim=-2,  keepdim=True) * \
```
Comment (ExponentialML) [inline stable_lora/lora.py:187]: This may be a hack, and while it does indeed work, I'm unsure if this is the correct way to do this here. 

The original shape for ModelScope's Conv3d has a kernel_size of `(3, 1, 1)`, which causes errors when trying to do batch matrix multiplication when adding weights on the forward pass. 

We set a shape of `(in_channels, out_channels, kernel_size, **kernel_size**, 1)` when we initialize the Conv3d LoRA. This allows us to do bmm correctly.

Afterwards during the forward pass, we simply mean over `dim=-2` while keeping the shape. Since this dimension has a size of 1 initially, the weight updates should not be affected negatively as the other shapes are still intact.

It may be useful to add a warning here until there is confirmation that this is correct, but I would consider this passable as it doesn't seem to cause any negative impact.
